### ISSUE TITLE
[Feature request] Add an option to forbid C++ method argument indentation.

### ISSUE THREAD

**Issue by liutikas** [checkstyle/checkstyle#354]
Checkstyle indentation module should have an option to forbid C++ method argument indentation.

Here is an example of a failing class:
```java
public class MyClass {
    public MyClass(int appleCount,
                   int bananaCount,
                   int pearsCount) {
    }
}
```

Checkstyle config used:
```xml
<module name="Checker">
  <module name="TreeWalker">
    <module name="Indentation">
      <property name="severity" value="error"/>
      <property name="basicOffset" value="4"/>
      <property name="throwsIndent" value="8"/>
      <property name="lineWrappingIndentation" value="8"/>
    </module>
  </module>
</module>
```

Expected warnings:
```
MyClass.java:3: Method parameter has incorrect indentation level 19, expected level should be 8.
MyClass.java:4: Method parameter has incorrect indentation level 19, expected level should be 8.
```

Expected correct code formatting:
```java
public class MyClass {
    public MyClass(int appleCount,
        int bananaCount,
        int pearsCount) {
    }
}
```

**Comment by romani** [comment]
fixed, will be in 6.3

**Comment by liutikas** [comment]
I found some issues when using this new rule in Checkstyle 6.5

```xml
<module name="Checker">
  <module name="TreeWalker">
    <module name="Indentation">
      <property name="severity" value="error"/>
      <property name="basicOffset" value="4"/>
      <property name="throwsIndent" value="8"/>
      <property name="lineWrappingIndentation" value="8"/>
      <property name="forceStrictCondition" value="true"/>
    </module>
  </module>
</module>
```

The following code is expected to be correctly indented, but I am getting warnings

```java
public class MyClass {
    public MyClass() {
        callMe(ThisReallyLongClassThatNeedsALongName.variableA,
                ThisReallyLongClassThatNeedsALongName.variableB
                        && ThisReallyLongClassThatNeedsALongName.variableC);
        ThisReallyLongClassThatNeedsALongName
                .thisReallyLongMethodNameThatHappensToWrap(
                        ThisReallyLongClassThatNeedsALongName.variableA);
        if (ThisReallyLongClassThatNeedsALongName.variableA
                && ThisReallyLongClassThatNeedsALongName.isEqual(
                        ThisReallyLongClassThatNeedsALongName2)) {
            return;
        }
    }
}
```

These are the warnings:
```
MyClass.java:5: '&&' have incorrect indentation level 24, expected level should be 16.
MyClass.java:8: 'ThisReallyLongClassThatNeedsALongName' have incorrect indentation level 24, expected level should be 16.
MyClass.java:11: 'ThisReallyLongClassThatNeedsALongName2' have incorrect indentation level 24, expected level should be 16.
```

Could we reopening this issue to fix this?

**Comment by liutikas** [comment]
Any updates on this?

**Comment by romani** [comment]
no , we are in middle of big update https://groups.google.com/forum/#!topic/checkstyle-devel/kDGBLXpUohk for that Check, but I put a hight demand on this issue.

**Comment by marlowa** [comment]
> That sounds good. This new additional option is exactly what we want.

I disagree. I would want a formatter/checker to work the way pylint does and pylint accepts it when the args line up. Of course it can only accept it when spaces are used with no tabs.

**Comment by romani** [comment]
some attempt was done at https://github.com/checkstyle/checkstyle/pull/19603 but not finished, everyone is welcome to continue

**Comment by ZaheerAhmadDev** [comment]
I am on it.

**Comment by romani** [comment]
we already have https://checkstyle.org/checks/misc/indentation.html#forceStrictCondition

@ZaheerAhmadDev , please share CLI output how it works or issue author code snippet.

**Comment by ZaheerAhmadDev** [comment]
```
zaheerahmad@fedora:~/checks$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="Indentation">
      <property name="severity" value="error"/>
      <property name="basicOffset" value="4"/>
      <property name="throwsIndent" value="8"/>
      <property name="lineWrappingIndentation" value="8"/>
      <property name="forceStrictCondition" value="true"/>
    </module>
  </module>
</module>
```
```java
zaheerahmad@fedora:~/checks$ cat src/Test.java
public class Test {
    public Test() {
        callMe(ThisReallyLongClassThatNeedsALongName.variableA,
                ThisReallyLongClassThatNeedsALongName.variableB
                        && ThisReallyLongClassThatNeedsALongName.variableC);
        ThisReallyLongClassThatNeedsALongName
                .thisReallyLongMethodNameThatHappensToWrap(
                        ThisReallyLongClassThatNeedsALongName.variableA);
        if (ThisReallyLongClassThatNeedsALongName.variableA
                && ThisReallyLongClassThatNeedsALongName.isEqual(
                        ThisReallyLongClassThatNeedsALongName2)) {
            return;
        }
    }
}
```

```
zaheerahmad@fedora:~/checks$ java -jar checkstyle-14.0.0-all.jar -c config.xml  src/Test.java
Starting audit...
[ERROR] /home/zaheerahmad/checks/src/Test.java:5:25: '&&' has incorrect indentation level 24, expected level should be 16. [Indentation]
[ERROR] /home/zaheerahmad/checks/src/Test.java:8:25: 'ThisReallyLongClassThatNeedsALongName' has incorrect indentation level 24, expected level should be 16. [Indentation]
[ERROR] /home/zaheerahmad/checks/src/Test.java:11:25: 'ThisReallyLongClassThatNeedsALongName2' has incorrect indentation level 24, expected level should be 16. [Indentation]
Audit done.
Checkstyle ends with 3 errors.
```
code is correctly indented `forceStrictCondition` when true it should not show these violation.

**Comment by romani** [comment]
please share on:
```
public class MyClass {
    public MyClass(int appleCount,
                   int bananaCount,
                   int pearsCount) {
    }
}
```

**Comment by ZaheerAhmadDev** [comment]
```
zaheerahmad@fedora:~/checks$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="Indentation">
      <property name="severity" value="error"/>
      <property name="basicOffset" value="4"/>
      <property name="throwsIndent" value="8"/>
      <property name="lineWrappingIndentation" value="8"/>
      <property name="forceStrictCondition" value="true"/>
    </module>
  </module>
</module>
```
```java
zaheerahmad@fedora:~/checks$ cat src/MyClass.java
public class MyClass {
    public MyClass(int appleCount,
                   int bananaCount,
                   int pearsCount) {
    }
}
```

```
zaheerahmad@fedora:~/checks$ java -jar checkstyle-14.0.0-all.jar -c config.xml  src/MyClass.java
Starting audit...
[ERROR] /home/zaheerahmad/checks/src/MyClass.java:3:20: 'int' has incorrect indentation level 19, expected level should be 12. [Indentation]
[ERROR] /home/zaheerahmad/checks/src/MyClass.java:4:20: 'int' has incorrect indentation level 19, expected level should be 12. [Indentation]
Audit done.
Checkstyle ends with 2 errors.
```

**Comment by romani** [comment]
thanks a lot .

issue is resovled, user should update config to get what he needs in behavior.
